src/simulator/src/utils.js

```diff
@@ -269,50 +248,6 @@ export function parseNumber(num) {
     return parseInt(num)
 }
 
-export function setupBitConvertor() {
-    console.log('check bit convertor')
-    $('#decimalInput').on('keyup', function () {
```

Comment:
we lost the functionality of rejecting wrong values 
eg: in binary if we add anything else then `0`,`1` it gets removed 

[Screencast from 2024-01-04 10-25-27.webm](https://github.com/CircuitVerse/cv-frontend-vue/assets/96580571/66dcec26-3ce3-42a7-8b27-463329b92fee)

Reply:
we can use the old method of `parseInt` or better we can use `regex` and have the functionality

Reply:
```js
function converter(e: KeyboardEvent) {
    const target = <HTMLInputElement>e.target!
    let value = target.value;

    // Regular expressions for validating input
    const regexBinary = /[^01]/g;
    const regexOctal = /[^0-7]/g;
    const regexDecimal = /[^0-9]/g;
    const regexHex = /[^0-9A-Fa-f]/g;

    switch (target.id) {
        case 'decimalInput':
            value = value.replace(regexDecimal, '');
            decimalConverter(value);
            break;
        case 'binaryInput':
            value = value.replace(regexBinary, '');
            binaryConverter(value);
            break;
        case 'bcdInput':
            value = value.replace(regexBinary, '');
            bcdConverter(value);
            break;
        case 'octalInput':
            value = value.replace(regexOctal, '');
            octalConverter(value);
            break;
        case 'hexInput':
            value = value.replace(regexHex, '');
            hexConverter(value);
            break;
    }

    // Update the input field with the cleaned value
    target.value = value;
}
```

use this